lux-lib/src/path.rs

```diff
@@ -63,9 +63,19 @@ impl Paths {
                 Ok::<Paths, TreeError>(paths)
             })?;
 
-        let lib_path = option_env!("LUX_LIB_DIR")
-            .map(PathBuf::from)
-            .unwrap_or_else(|| Config::get_default_data_path().unwrap())
+        let lib_name = format!("lux-lua{}", tree.version());
+        let lib_path: PathBuf = pkg_config::Config::new()
+            .print_system_libs(false)
+            .cargo_metadata(false)
+            .env_metadata(false)
+            .probe(&lib_name)
+            .ok()
+            .and_then(|library| library.link_paths.first().cloned())
+            .unwrap_or(
+                option_env!("LUX_LIB_DIR")
+                    .map(PathBuf::from)
+                    .unwrap_or_else(|| Config::get_default_data_path().unwrap()),
+            )
```

Comment:
question: shouldn't we prioritise `LUX_LIB_DIR`, since that's a compile time constant that's injected into the binary at build time? If the packager decided to set it, they probably would want it to be respected at all times.

Reply:
Good point.